src/binder/assemblybinder.cpp

```diff
@@ -685,19 +685,23 @@ namespace BINDER_SPACE
 
         StackSString sCoreLib;
 
-        // At run-time, System.Private.CoreLib.ni.dll is typically always available, and
-        // System.Private.CoreLib.dll is typically not.  So check for the NI first.
+        // At run-time, System.Private.CoreLib.dll is expected to be the NI image.
+        // Try binding to it first as that. If it fails, we will try to bind to it as 
+        // IL image.
         sCoreLib = sCoreLibDir;
-        sCoreLib.Append(CoreLibName_NI_W);
+        sCoreLib.Append(CoreLibName_IL_W);
+#if defined(FEATURE_PAL) && !defined(_TARGET_AMD64_)
+        // Native image for core library only exists for X64 on non-Windows. As other architectures
+        // bring them up, we should update this condition.
+        fBindToNativeImage = false;
```

Comment:
@jkotas I had to add this for Linux-Arm builds. My observation is that if the first call to GetAssembly below attempts to bind to SPC.dll (which is IL binary on Linux-Arm since it is not crossgenned yet), it likely results in a failure and thus, CLR startup fails.

Setting this to false makes us attempt to use the inner GetAssembly call that attempts to bind to IL assembly with fExplicitBindToNativeImage as FALSE. And this, approach appears to work (and is what was prior to my change as well since spc.ni.dll didnt exist for Arm).

I dont have a way to setup a repro for running Linux Arm tests, so I adopted setting the flag to false for now though I dont really like it though.

Do you have ideas on why the failure maybe happening since the failure case (bind to IL SPC.dll as with fExplicitBindToNativeImage  as TRUE) works on Windows?